## Introduction

Genes operate within specific networks that are tailored to different functions, as evidenced by the research highlighting the complex interplay between genetic variations and their roles in biological processes relevant to diseases [@pmid:19104045; @doi:10.1038/ng.3259; @doi:10.1038/s41467-018-06022-6].
A deeper understanding of how these variations influence gene expression across various cell types is essential for elucidating the mechanisms behind complex traits and developing more effective treatments [@doi:10.1126/science.aaz1776; @doi:10.1038/s41586-020-2559-3; @doi:10.1038/s41576-019-0200-9; @doi:10.1038/ng.3314; @doi:10.1371/journal.pgen.1008489].
Previous studies have mapped numerous regulatory DNA elements and explored genetic impacts on gene expression across different tissues, which enhances our understanding of gene regulation [@doi:10.1038/nature11247; @doi:10.1038/nature14248; @doi:10.1038/nature12787; @doi:10.1038/s41586-020-03145-z; @doi:10.1126/science.aaz1776].
By integrating functional genomics with Genome-Wide Association Studies (GWAS) data, researchers have advanced the identification of key transcriptional mechanisms.
These mechanisms are crucial for understanding how disruptions in gene regulation can lead to pathology that varies by tissue type and cell lineage [@doi:10.1038/s41588-018-0081-4; @doi:10.1016/j.ajhg.2018.04.002; @doi:10.1038/ncomms6890; @pmid:20624743; @pmid:14707169; @doi:10.1073/pnas.0810772105].
This manuscript seeks to fill the knowledge gap by further dissecting these complex genetic interactions, aiming to clarify their implications for targeted therapeutic strategies.


The increasing availability of gene expression data from diverse tissues has provided a robust foundation for understanding biological processes at the molecular level [@doi:10.1038/nbt.3838; @doi:10.1038/s41467-018-03751-6; @doi:10.1126/science.aaz1776; @doi:10.1186/s13040-020-00216-9].
One powerful tool that leverages this data is the transcription-wide association study (TWAS), which utilizes expression quantitative trait loci (eQTLs) to elucidate mechanisms underlying genome-wide association study (GWAS) findings.
TWAS has proven effective in linking gene regulatory disruptions to human diseases [@doi:10.1371/journal.pgen.1009482; @doi:10.1038/ng.3506; @doi:10.1371/journal.pgen.1007889; @doi:10.1038/ng.3367], enhancing our understanding of disease etiology at the transcriptome level [@pmid:33931583; @doi:10.1101/2021.10.21.21265225; @pmid:31036433].
This approach has not only advanced disease-risk prediction through polygenic scores [@doi:10.1186/s13059-021-02591-w] but also aided in drug repurposing efforts [@doi:10.1038/nn.4618].
However, the primary limitation of TWAS is its focus on individual genes, which overlooks the broader, potentially more complex interactions occurring within gene networks.
This gap highlights the need for methodologies that can capture and analyze these intricate network-level interactions.


Understanding the complex interactions between genes is pivotal in elucidating the genetic architecture of complex traits.
The omnigenic model posits that nearly all genes expressed in disease-relevant cells potentially influence these traits, necessitating sophisticated analytical methods to unravel these intricate relationships [@doi:10.1016/j.cell.2017.05.038].
This model is supported by evidence of widespread gene pleiotropy, which underscores the highly interconnected nature of transcriptional networks [@doi:10.1038/s41588-019-0481-0; @doi:10.1038/ng.3570].
A promising approach to deciphering these interactions involves the use of gene modules, which are clusters of genes that exhibit similar expression patterns across various conditions [@pmid:22955619; @pmid:25344726; @doi:10.1038/ng.3259].
Techniques such as matrix factorization, including independent and principal component analysis (ICA/PCA), have proven effective in identifying these modules by analyzing co-expression patterns, thereby capturing local expression effects and managing overlaps in module data [@doi:10.1038/s41467-018-03424-4].
The integration of genetic studies with these unsupervised learning-derived gene modules could significantly advance our understanding of the mechanisms underlying disease onset and progression [@pmid:25344726; @pmid:18631455].
This manuscript aims to bridge this knowledge gap by further refining these integrative methods to enhance our comprehension of disease mechanisms.


In this study, we introduce PhenoPLIER, an innovative approach that leverages the omnigenic model to enhance the analysis of genetic studies through a gene module framework.
Our method uniquely combines various data types, including gene-trait associations and drug transcriptional profiles, into a unified analysis.
Specifically, we incorporate gene-trait associations from PhenomeXcan [@doi:10.1126/sciadv.aba2083] and drug response data from LINCS L1000 [@doi:10.1016/j.cell.2017.10.049] into a cohesive latent space constructed from extensive public gene expression datasets, such as recount2 [@doi:10.1016/j.cels.2019.04.003; @doi:10.1038/nbt.3838].
By employing matrix factorization techniques [@doi:10.1038/s41592-019-0456-1; @doi:10.1016/j.cels.2019.04.003], PhenoPLIER effectively identifies and maps gene modules, highlighting significant associations with diseases in relevant cell typesâ€”ranging from T cells in hypothyroidism to various brain cells in neuropsychiatric disorders.
Notably, our results demonstrate that gene modules provide a more robust framework for understanding complex traits and predicting drug-disease interactions compared to traditional single-gene analyses.
This module-based perspective not only confirms known associations but also uncovers new candidate targets that may not be evident through conventional methods.
Ultimately, PhenoPLIER represents a shift towards a more nuanced interpretation of genetic data, emphasizing the collective role of gene networks in disease mechanisms and therapeutic interventions.
