### PhenoPLIER: an integration framework based on gene co-expression patterns

![
**Schematic of the PhenoPLIER framework.**
**a)** High-level schematic of PhenoPLIER (a gene module-based method) in the context of TWAS (single-gene) and GWAS (single-variant).
In GWAS, we identify variants associated with traits.
In TWAS, first, we identify variants that are associated with gene expression levels (eQTLs); then, prediction models based on eQTLs are used to impute gene expression, which is used to compute gene-trait associations.
Resources such as LINCS L1000 provide information about how a drug perturbs gene expression; at the bottom-right corner, we show how a drug downregulates two genes (A and C).
In PhenoPLIER, these data types are integrated using groups of genes co-expressed across one or more conditions (such as cell types) that we call gene modules or latent variables/LVs. Created with BioRender.com.
**b)** The integration process in PhenoPLIER uses low-dimensional representations (matrices $\mathbf{Z}$ and $\mathbf{B}$) learned from large gene expression datasets (top).
We used gene-drug information $\mathbf{L}$ from LINCS L1000 and gene-trait associations $\mathbf{M}$ from TWAS: PhenomeXcan was used as the discovery cohort, and eMERGE as replication (middle).
PhenoPLIER provides three computational components (bottom):
1) an LV-based regression model that associates an LV $j$ ($\mathbf{Z}_j$) with a trait $i$ ($\mathbf{M}_i$),
2) a clustering framework that learns groups of traits from TWAS associations projected into the LV space ($\hat{\mathbf{M}}$),
and 3) an LV-based drug repurposing approach that uses the projection of TWAS ($\hat{\mathbf{M}}$) and LINCS L1000 ($\hat{\mathbf{L}}$) into the LV space.
**c)** Genes that are part of LV603, termed as a neutrophil signature [@doi:10.1016/j.cels.2019.04.003], were expressed in relevant cell types (top), with 53 independent samples expressed in Neutrophils, 59 in Granulocytes, and 20 in Whole blood, 56 in PBMC, 8 in mDCs, 29 in Monocytes, and 5 in Epithelial cells (the boxplot shows the 25th, 50th and 75th percentiles while the whiskers extend to the minimum/maximum values).
LV603 was associated in PhenoPLIER with neutrophil counts and other white blood cells (bottom, showing the top 10 traits for LV603 after projecting gene-trait associations in PhenomeXcan).
eQTLs: expression quantitative trait loci;
MVN: multivariate normal distribution;
PBMC: peripheral blood mononuclear cells;
mDCs: myeloid dendritic cells.
](images/entire_process/entire_process.svg "PhenoPLIER framework"){#fig:entire_process width="100%"}


To determine the effectiveness of PhenoPLIER in integrating gene-trait and gene-drug associations with gene modules expressed in specific contexts, we employed a flexible computational framework.
PhenoPLIER utilizes a latent representation, with latent variables (LVs) representing gene modules, derived from a large gene expression compendium (Figure {@fig:entire_process}b, top).
This framework integrates Transcriptome-Wide Association Studies (TWAS) with drug-induced transcriptional responses (Figure {@fig:entire_process}b, middle) for a comprehensive joint analysis.
The approach involves three main components (Figure {@fig:entire_process}b, bottom, see [Methods](#sec:methods)): 1) an LV-based regression model to compute associations between LVs and traits, 2) a clustering framework to identify groups of traits with shared transcriptomic properties, and 3) an LV-based drug repurposing strategy that links diseases to potential treatments.
To validate our regression model and clustering framework, we conducted extensive simulations ([Supplementary Note 1](#sm:reg:null_sim); [Supplementary Note 2](#sm:clustering:null_sim)), confirming proper calibration and expected results under a model of no association.


To determine the robustness and replicability of our findings, we employed TWAS results from PhenomeXcan [@doi:10.1126/sciadv.aba2083] and the eMERGE network [@doi:10.1101/2021.10.21.21265225] as discovery and replication cohorts, respectively ([Methods](#sec:methods:twas)).
PhenomeXcan offers gene-trait associations for 4,091 diseases and traits derived from the UK Biobank [@doi:10.1038/s41586-018-0579-z] and other studies, while eMERGE analyses were conducted across 309 phecodes.
TWAS results were generated using two statistical methods: Summary-MultiXcan (S-MultiXcan) associations for regression and clustering, and Summary-PrediXcan (S-PrediXcan) associations for drug repurposing (see [Methods](#sec:methods:predixcan)).
Additionally, we incorporated colocalization results to estimate the probability of overlap between GWAS and eQTL signals.
For the drug-repurposing analysis, we utilized transcriptional responses to small molecule perturbations from LINCS L1000 [@doi:10.1016/j.cell.2017.10.049], which includes data on 1,170 compounds.
These comprehensive approaches confirm the validity and reproducibility of our findings across different datasets and methodologies.


To determine whether the latent gene expression representation could be effectively derived, we utilized the MultiPLIER models [@doi:10.1016/j.cels.2019.04.003].
These models were created by applying the PLIER method [@doi:10.1038/s41592-019-0456-1], a matrix factorization technique, to the recount2 dataset [@doi:10.1038/nbt.3838].
Recount2 is a uniformly curated compilation of transcript-level gene expression data obtained via RNA-seq from a diverse array of samples encompassing various disease states, cell differentiation stages, and stimuli (see [Methods](#sec:methods:multiplier)).
The MultiPLIER models successfully extracted 987 latent variables (LVs) by optimizing both data reconstruction and the alignment of LVs with existing knowledge and pathways.
Thus, the latent gene expression representation was effectively obtained using the MultiPLIER models.


To investigate whether latent variables (LVs) represent functional sets of genes regulated by the same transcriptional program, we projected Transcriptome-Wide Association Studies (TWAS) and pharmacologic perturbation data into this latent space [@doi:10.1186/1471-2164-7-187; @doi:10.1186/s13059-019-1835-8].
PhenoPLIER was utilized to convert gene associations into an LV score by multiplying all genes' standardized effect sizes for a trait (from TWAS) or differential expression values for a drug (from pharmacologic perturbation data) by the LV genes' weights and summing the results.
This method links traits and drugs to functionally-related groups of genes or LVs, rather than individual genes.
PhenoPLIER also uses LV annotations about specific conditions where gene groups are expressed, such as cell types, tissues, developmental stages, disease stages, or under distinct stimuli, which can enhance result interpretability.
MultiPLIER's models provide this information by linking LVs to samples annotated for experimental conditions (represented by matrix $\mathbf{B}$ in Figure {@fig:entire_process}b).
For instance, in the original MultiPLIER study, LV603 was associated with a known neutrophil pathway and correlated with neutrophil count estimates from whole blood RNA-seq profiles [@doi:10.1186/s13059-016-1070-5].
Our analysis of LV603 using PhenoPLIER revealed that neutrophil counts and other white blood cell traits ranked among the top 10 traits out of 4,091 (Figure {@fig:entire_process}c, bottom), and basophils count and percentage were significantly associated with this LV (Table @tbl:sup:phenomexcan_assocs:lv603).
Additionally, LV603's genes were expressed in highly relevant cell types (Figure {@fig:entire_process}c, top).
These findings suggest that functionally related and co-expressed gene groups correspond to trait-associated genes, linking transcriptional mechanisms from diverse datasets to complex traits.


To determine the utility of PhenoPLIER in addressing specific biological questions, we investigated whether disease-associated genes are part of modules expressed in particular tissues and cell types.
We also explored if these cell type-specific modules are linked to various diseases, which could illustrate "network pleiotropy" from an omnigenic perspective [@doi:10.1016/j.cell.2017.05.038].
Additionally, we examined whether a subset of module genes aligns closely with the definition of "core" genes—those that directly affect the trait without mediating the regulation of other genes [@doi:10.1016/j.cell.2019.04.014]—to identify alternative, potentially superior candidate targets.
Finally, we assessed whether drugs perturb these transcriptional mechanisms and if they can suggest potential mechanisms of action.
Our findings demonstrate that PhenoPLIER effectively addresses these questions, thereby confirming its utility in identifying biologically relevant gene modules and potential therapeutic targets.
