### LVs predict drug-disease pairs better than single genes

To determine how substituting latent variables (LVs) for individual genes predicts known treatment-disease relationships, we utilized transcriptional responses to small molecule perturbations profiled in LINCS L1000 [@doi:10.1016/j.cell.2017.10.049], which were further processed and mapped to DrugBank IDs [@doi:10.1093/nar/gkt1068; @doi:10.7554/eLife.26726; @doi:10.5281/zenodo.47223].
Utilizing an established drug repurposing strategy that matches reversed transcriptome patterns between genes and drug-induced perturbations [@doi:10.1126/scitranslmed.3002648; @doi:10.1126/scitranslmed.3001318], we adopted a previously described framework that uses imputed transcriptomes from TWAS to prioritize drug candidates [@doi:10.1038/nn.4618].
Specifically, we computed a drug-disease score by calculating the negative dot product between the $z$-scores for a disease (from TWAS) and the $z$-scores for a drug (from LINCS) across sets of genes of different sizes (see [Methods](#sec:methods:drug)).
A large score for a drug-disease pair indicated that higher (lower) predicted expression values of disease-associated genes are down (up)-regulated by the drug, thus predicting a potential treatment.
For the LV-based approach, we estimated how pharmacological perturbations affected the gene module activity by projecting expression profiles of drugs into our latent representation (Figure {@fig:entire_process}b).
We evaluated the prediction performance using a manually-curated gold standard set of drug-disease medical indications [@doi:10.7554/eLife.26726; @doi:10.5281/zenodo.47664] for 322 drugs across 53 diseases.
This approach demonstrated that substituting LVs for individual genes can effectively predict known treatment-disease relationships.


![
**Drug-disease prediction performance for gene-based and LV-based approaches.**
The receiver operating characteristic (ROC) (left) and the precision-recall curves (right) for a gene-based and LV-based approach.
"Random" refers to the average precision of a hundred classifiers with randomly permuted scores, where the error band represents the 95% confidence interval.
AUC: area under the curve; AP: average precision.
](images/drug_disease_prediction/roc_pr_curves.svg "ROC-PR curves for drug-disease prediction"){#fig:drug_disease:roc_pr width="80%"}


To assess whether the latent space representation effectively captures biologically meaningful patterns, we compared the performance of the LV-based method to the gene-based method.
Despite the inherent information loss in projecting gene-trait associations and drug-induced expression profiles into a compressed latent space, the LV-based method demonstrated superior performance with an area under the curve (AUC) of 0.632 and an average precision of 0.858 (Figure @fig:drug_disease:roc_pr).
These results indicate that the low-dimensional space retains critical biological information, linking pathophysiological processes with drug mechanisms of action.


To determine the biological plausibility of the latent variables (LVs) driving the prediction, we examined a specific drug-disease pair involving nicotinic acid (niacin).
Niacin, a B vitamin, is widely used to treat lipid disorders, though its efficacy in preventing cardiovascular disease is debated [@pmid:22085343; @pmid:25014686; @pmid:30977858].
It impacts multiple tissues, but its mechanisms remain incompletely understood [@doi:10.1016/j.amjcard.2008.02.029; @doi:10.1194/jlr.S092007; @pmid:24363242; @pmid:24713591].
Niacin increases high-density lipoprotein (HDL) by inhibiting an HDL catabolism receptor in the liver and decreases low-density lipoproteins (LDL) by inhibiting diacylglycerol acyltransferaseâ€“2 (DGAT2), affecting triglyceride synthesis in hepatocytes or adipocyte triglyceride lipolysis [@doi:10.1016/j.amjcard.2008.02.029].
In our study, niacin was included in the gold standard set for atherosclerosis (AT) and coronary artery disease (CAD).
It was predicted as a medical indication for CAD by both gene-based and LV-based approaches, with scores of 0.51 and 0.96, respectively.
For AT, the LV-based approach predicted niacin as a therapeutic drug with a score of 0.52, while the gene-based method gave a negative score of -0.01.
Since LVs represent interpretable features linked to specific cell types, we identified which LVs positively contributed to these predictions.
Notably, LV246 (Figure @fig:lv246), expressed in adipose tissue and liver and associated with plasma lipids and high cholesterol (Table @tbl:sup:phenomexcan_assocs:lv246), was the 16th most important module in predicting niacin as a therapeutic drug for AT.
Additionally, LV246 ranked highly for other cardiovascular diseases, such as ischaemic heart disease (15th module) and high cholesterol (7th module).
Thus, our analysis supports the biological plausibility of the LVs in predicting niacin's therapeutic potential for cardiovascular conditions.

![
**Top cell types/tissues where LV116's genes are expressed.**
Values in the $y$-axis come from matrix $\mathbf{B}$ in the MultiPLIER models (Figure {@fig:entire_process}b).
In the $x$-axis, cell types/tissues are sorted by the maximum sample value.
The figure shows a clear immune response with cell types under different stimuli.
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP045500 -->
MS: multiple sclerosis;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP045569 -->
<!-- PBMCs: peripheral blood mononuclear cells; -->
HSV: treated with herpes simplex virus;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP015670 -->
WNV: infected with West Nile virus;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP062958 -->
IFNa: treated with interferon-alpha;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP039361 -->
HMDM: human peripheral blood mononuclear cell-derived macrophages;
<!-- IPSDM: human induced pluripotent stem cell-derived macrophages; -->
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP056733 -->
Salm: infected with *Salmonella typhimurium*;
Yers: infected with *Yersinia pseudotuberculosis*;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP062966 -->
ISM: Interferon Signature Metric;
SLE: Systemic lupus erythematosus.
](images/lvs_analysis/lv116/lv116-cell_types.svg "LV116 cell types"){#fig:lv116:cell_types width="100%"}



To explore the mechanisms of action of niacin beyond its known effects, we analyzed other top niacin-contributing latent variables (LVs) across different cardiovascular diseases.
Specifically, *GPR109A/HCAR2* encodes a G protein-coupled high-affinity niacin receptor in various immune cells, including monocytes, macrophages, neutrophils, and dendritic cells [@doi:10.1016/j.tips.2006.05.008; @doi:10.1038/sj.jid.5700586].
Initially, the antiatherogenic effects of niacin were attributed solely to the inhibition of lipolysis in adipose tissue.
However, recent studies have demonstrated that nicotinic acid can reduce atherosclerosis progression independently of its antidyslipidemic activity by activating *GPR109A* in immune cells, thereby enhancing anti-inflammatory processes [@doi:10.1172/JCI41651; @doi:10.1161/ATVBAHA.108.179283].
Additionally, the common adverse effect of flushing is caused by the activation of GPR109A in Langerhans cells.
By examining the cell types where top-contributing modules are expressed, such as LV116 and LV931, we observed strong signatures in monocytes, macrophages, neutrophils, and dendritic cells (Figures @fig:lv116:cell_types and @fig:sup:lv931; Tables @tbl:sup:multiplier_pathways:lv116 and @tbl:sup:multiplier_pathways:lv931).
For example, LV116's genes are involved in immune responses to various stimuli, including pathogens causing diarrhea [@doi:10.1371/journal.pone.0192082], multiple sclerosis, systemic lupus erythematosus [@doi:10.1371/journal.pone.0109760; @doi:10.1126/science.aac7442], and viral infections like herpes simplex [@url:https://www.ncbi.nlm.nih.gov/bioproject/PRJNA258384], West Nile virus [@doi:10.3390/v5071664], and *Salmonella typhimurium* [@doi:10.1038/srep16882].
These findings indicate that LV246, LV116, and LV931 are among the top 20 modules contributing to niacin prediction across different cardiovascular traits (Table @tbl:niacin:cardio:top_lvs), suggesting alternative mechanisms for niacin's effects.


<!-- niacin:cardiovascular:top_lvs:start DISABLE NOW, BUT HAS TO BE ADDED THE "end" version just below to update table -->
| LV    | Cell type             | Disease                                      |
|:------|:----------------------|:---------------------------------------------|
| LV116 | Immune cells, skin    | Atherosclerosis (ICD10 I70)                  |
|       |                       | Chronic ischaemic heart disease (ICD10 I25)  |
|       |                       | Heart attack, angina, stroke or hypertension |
|       |                       | Ischaemic heart disease (wide definition)    |
| LV931 | Immune cells          | Atherosclerosis (ICD10 I70)                  |
|       |                       | Heart attack, angina, stroke or hypertension |
|       |                       | Ischaemic heart disease (wide definition)    |
| LV246 | Adipose tissue, liver | Atherosclerosis (ICD10 I70)                  |
|       |                       | High cholesterol (self-reported)             |
|       |                       | Ischaemic heart disease (wide definition)    |

Table: **LVs among the top 20 contributors to the prediction of niacin for five cardiovascular diseases.** "Heart attack, angina, stroke or hypertension" refers to the UK Biobank data-field 6150. GWAS sample size: Atherosclerosis (361,194 in total and 566 cases), Chronic ischaemic heart disease (361,194 in total and 12,769 cases), Heart attack, angina, stroke or hypertension (360,420 in total and 253,565 cases), Ischaemic heart disease/wide definition (361,194 in total and 20,857 cases), High cholesterol/self-reported (361,141 in total and 43,957 cases).  {#tbl:niacin:cardio:top_lvs}


To explore additional mechanisms by which niacin may exert its effects, we examined other latent variables (LVs) beyond cardiovascular traits.
One such LV, LV66, emerged as significantly influenced by niacin treatment (Figure @fig:sup:lv66) and was predominantly expressed in ovarian granulosa cells.
Recent studies have identified this compound as a potential therapeutic agent for ovarian diseases [@doi:10.1159/000495051; @doi:10.1071/RD20306], demonstrating its ability to promote follicle growth and prevent granulosa cell apoptosis in animal models.
These findings suggest that niacin's impact extends to ovarian health, highlighting its broader therapeutic potential.
