### LVs reveal trait clusters with shared transcriptomic properties

![
**Cluster analysis on traits using the latent gene expression representation.**
**a)** The projection of TWAS results on 3,752 traits into the latent gene expression representation is the input data to the clustering process.
A linear (PCA) and non-linear (UMAP) dimensionality reduction techniques were applied to the input data, and five different clustering algorithms processed all data versions.
These algorithms derive partitions from the data using different parameters (such as the number of clusters), leading to an ensemble of 4,428 partitions.
Then, a distance matrix is derived by counting how many times a pair of traits was grouped in different clusters across the ensemble.
Finally, a consensus function is applied to the distance matrix to generate consolidated partitions with different numbers of clusters (from 2 to $\sqrt{n}\approx$ 60).
These final solutions were represented in the clustering tree (Figure @fig:clustering:tree).
**b)** The clusters found by the consensus function were used as labels to train a decision tree classifier on the original input data, which detects the LVs that better differentiate groups of traits.
](images/clustering/clustering_design.svg "Cluster analysis on traits"){#fig:clustering:design width="100%"}


To explore how different gene-trait associations group together based on transcriptional processes, we implemented a consensus clustering framework.
This approach is necessary due to the inherent biases of individual clustering algorithms, which make different assumptions about data structure.
Our method integrates various clustering solutions that categorize traits using distinct algorithms, as detailed in our [Methods](#sec:methods:clustering) section.
Such ensemble techniques are advocated to mitigate common errors in cluster analysis of biological data, as supported by literature [@doi:10.1126/scisignal.aad1932].
To ensure diversity, essential for the robustness of consensus methods, we processed multiple data variants through different algorithms with varied parameters (Figure {@fig:clustering:design}a).
A consensus function then merges these into a unified clustering solution, which typically surpasses the performance of any single algorithm [@Strehl2002; @doi:10.1109/TPAMI.2005.113].
Our approach yielded 15 final consensus clustering solutions, with the number of clusters ranging from 5 to 29, determined by the highest concordance within the ensemble [@Strehl2002] (Figure @fig:sup:clustering:agreement).
Rather than selecting a single final solution, we utilized a clustering tree to investigate stable trait groups across various resolutions [@doi:10.1093/gigascience/giy083] (Figure @fig:clustering:tree).
To discern the latent variables that differentiate these trait groups, we trained a decision tree classifier using the identified clusters as labels, based on the input data $\hat{\mathbf{M}}$ (Figure {@fig:clustering:design}b, see [Methods](#sec:methods:clustering)).
This comprehensive approach not only clarifies the structure of gene-trait associations but also enhances the reliability of identifying transcriptional processes that govern these associations.


![
**Clustering tree using multiple resolutions for clusters of traits.**
Each row represents a partition/grouping of the traits, and each circle is a cluster from that partition.
The number of clusters goes from 5 to 29.
Arrows indicate how traits in one cluster move across clusters from different partitions.
Most of the clusters are preserved across different resolutions, showing highly stable solutions even with independent runs of the clustering algorithm.
<!--  -->
RDW: red cell (erythrocyte) distribution width;
BMI: body mass index;
WC: waist circumference;
HC: hip circumference;
RA: rheumatoid arthritis;
SLE: systemic lupus erythematosus;
HTN: Hypertension;
IBD: inflammatory bowel disease;
SCZ: Schizophrenia;
CAD: Coronary artery disease;
AD: Alzheimer's disease;
<!--  -->
The full lists of traits in each cluster in the last five partitions of the tree (from $k=16$ to $k=29$) are in Supplementary Data 3-7.
](images/clustering/clustering_tree.svg "Clustering tree on groups of traits"){#fig:clustering:tree width="100%"}


To determine the stability and categorization of phenotypic traits in our study, we analyzed their clustering in a hierarchical tree structure as depicted in the figure [@fig:clustering:tree].
We identified five distinct branches at the first node: 0) a "large" branch encompassing a diverse array of traits like asthma, subjective well-being, and nutrient intake starting to subdivide at $k$=16; 1) heel bone densitometry measurements; 2) hematological assays on red blood cells; 3) physical measures such as spirometry and body impedance, along with anthropometric traits subdivided into fat-free and fat mass measures; and 4) a "complex" branch which includes keratometry measurements, assays on white blood cells and platelets, skin and hair color traits, autoimmune disorders, and cardiovascular diseases, with additional related traits like hand-grip strength [@pmid:25982160] and factors such as physical activity and diet.
The clustering results, detailed in Supplementary Data 3-7, showed high stability with the same traits frequently grouping together across different resolutions.
Notably, traits tended to migrate between clusters mainly within the "complex" branch and from the "large" to the "complex" branch, reflecting the common genetic and environmental underpinnings of complex diseases that challenge their classification into distinct clusters.
This analysis confirms the robustness of our clustering approach and the inherent complexity in categorizing multifaceted diseases.


![
**Cluster-specific and general transcriptional processes associated with different diseases.**
The plot shows a submatrix of $\hat{\mathbf{M}}$ for the main trait clusters at $k$=29, considering only LVs (rows) that are well-aligned with at least one pathway.
](images/clustering/global_clustermap-plain.svg "Heatmap with gene modules and traits"){#fig:clustering:heatmap width="100%"}


To understand the influence of latent variables (LVs) on trait clusters, we investigated which LVs were primarily responsible for driving these associations.
Utilizing decision tree classifiers, we analyzed the input data labeled by each cluster at $k$=29 (bottom of Figure @fig:clustering:tree) as detailed in the [Methods](#sec:methods:clustering) section.
This analysis identified key LVs that were most discriminative for each cluster.
Some of these LVs corresponded well with known biological pathways (Figure @fig:clustering:heatmap), while others, although not previously aligned with known pathways, were still expressed in relevant tissues (Figure @fig:sup:clustering:novel:heatmap).
The specificity of these LVs varied; for instance, some were highly specific to particular traits, whereas others were linked to a broader range of phenotypes, suggesting a role in general biological functions (Figure @fig:clustering:heatmap).
Further analysis using our regression framework confirmed significant associations between these LVs and various traits.
For example, LV928 and LV30, aligned with the early progenitors of the erythrocytes lineage [@doi:10.1016/j.cell.2011.01.004] and predominantly expressed during the early differentiation stages of erythropoiesis (Figures @fig:sup:lv928 and @fig:sup:lv30), showed strong associations with assays on red blood cells (FDR < 0.05; Tables @tbl:sup:phenomexcan_assocs:lv928, @tbl:sup:emerge_assocs:lv928, and @tbl:sup:emerge_assocs:lv30).
Conversely, LV730, expressed in thrombocytes from various cancer samples (Figure @fig:sup:lv730 and Table @tbl:sup:multiplier_pathways:lv730), and LV598, expressed in corneal endothelial cells (Figure @fig:sup:lv598 and Table @tbl:sup:multiplier_pathways:lv598), were each strongly associated with specific hematological and keratometry measurements, respectively (Tables @tbl:sup:phenomexcan_assocs:lv730 and @tbl:sup:phenomexcan_assocs:lv598).
This analysis confirms that certain LVs are significantly associated with specific traits, highlighting their potential roles in distinct biological pathways and functions.


To understand the specificity of latent variables (LVs) in relation to autoimmune and cardiovascular diseases, we analyzed the clustering of these diseases at a significant node where $k=10$ (middle of Figure @fig:clustering:tree).
We focused on two LVs, LV57 and LV844, to determine their disease associations and expression profiles.
LV57, prominently expressed in T cells (Figure @fig:sup:lv57 and Table @tbl:sup:multiplier_pathways:lv57), showed a strong correlation with autoimmune disorders in PhenomeXcan, including notable associations with hypothyroidism, which were confirmed in eMERGE (Table @tbl:sup:phenomexcan_assocs:lv57; Table @tbl:sup:emerge_assocs:lv57).
Interestingly, LV57 was also linked to deep venous thrombosis in both databases.
In contrast, LV844, which was expressed across various cell types including blood, breast organoids, myeloma cells, lung fibroblasts, and brain cells (Figure @fig:sup:lv844 and Table @tbl:sup:multiplier_pathways:lv844), demonstrated more specific associations with autoimmune conditions such as polymyalgia rheumatica, type 1 diabetes, rheumatoid arthritis, and celiac disease in PhenomeXcan (Table @tbl:sup:phenomexcan_assocs:lv844).
However, these associations did not replicate in eMERGE.
This analysis reveals that while LV57 impacts both autoimmune and cardiovascular diseases, LV844 shows a more targeted association with autoimmune disorders.


To investigate the association of specific latent variables (LVs) with cardiovascular and neuropsychiatric disorders, we analyzed data from PhenomeXcan and eMERGE databases.
Our analysis revealed 129 significant LV-trait associations within the cardiovascular sub-branch in PhenomeXcan and 23 in eMERGE.
Specifically, LV136, which is involved in collagen formation and muscle contraction pathways (Table @tbl:sup:multiplier_pathways:lv136), was linked to coronary artery disease and keratometry measurements in PhenomeXcan (Table @tbl:sup:phenomexcan_assocs:lv136) and to coronary atherosclerosis in eMERGE (phecode: 411.4) (Table @tbl:sup:emerge_assocs:lv136).
This LV was expressed across various cell types including fibroblasts and cardiomyocytes (Figure @fig:sup:lv136).
Additionally, we found associations between cardiovascular diseases and neuropsychiatric conditions such as Alzheimer's disease, schizophrenia, and attention deficit hyperactivity disorder (ADHD), which share common risk factors like hypertension and high cholesterol [@pmid:12093424; @doi:10.1161/CIRCULATIONAHA.113.002065; @doi:10.1192/bjp.bp.117.202606; @doi:10.1161/CIRCRESAHA.118.313563; @doi:10.1186/s12916-014-0206-2; @doi:10.1111/j.1076-7460.2007.06696.x].
Alzheimer's disease, in particular, showed a significant association with LV21 in PhenomeXcan (Table @tbl:sup:phenomexcan_assocs:lv21), a gene module not previously linked to known pathways but expressed in tissues such as monocytes/macrophages and aortic valves (Figure @fig:sup:lv21 and Table @tbl:sup:multiplier_pathways:lv21).
LV21 was also correlated with lipid levels and high cholesterol in both PhenomeXcan and eMERGE (Table @tbl:sup:emerge_assocs:lv21).
These findings suggest that shared transcriptional processes in specific tissues may underpin the link between cardiovascular and neuropsychiatric disorders, highlighting the importance of lipid metabolism in these diseases' pathogenesis, particularly through macrophage involvement in cholesterol transport and atherogenesis [@doi:10.1093/qjmed/hci136; @doi:10.3389/fphys.2020.00393].
