## Methods {#sec:methods}

### Methods

PhenoPLIER is a framework that integrates gene-trait associations and drug-induced transcriptional responses with groups of functionally related genes, termed gene modules or latent variables (LVs).
Gene-trait associations are computed using the PrediXcan family of methods, while latent variables are inferred through MultiPLIER models applied to extensive gene expression compendia.
PhenoPLIER offers the following components:

1.
**LV-Trait Association Regression Model**: This model computes associations between LVs and traits.
2.
**Consensus Clustering in Latent Space**: This approach identifies shared and distinct transcriptomic properties among traits.
3.
**Interpretable LV-Based Drug Repurposing Framework**: This framework facilitates the repurposing of drugs based on LVs.

Details of these methods are provided below.


### The PrediXcan family of methods for gene-based associations {#sec:methods:predixcan}

### Methods

We employed Summary-PrediXcan (S-PrediXcan) [@doi:10.1038/s41467-018-03621-1] and Summary-MultiXcan (S-MultiXcan) [@doi:10.1371/journal.pgen.1007889] as our gene-based statistical methods, which are part of the PrediXcan family of approaches [@doi:10.1038/ng.3367].
Collectively, we refer to these methods as transcription-wide association studies (TWAS).

S-PrediXcan, the summary-based version of PrediXcan, calculates the univariate association between a trait and the predicted expression of a gene in a single tissue.
Conversely, S-MultiXcan, the summary-based version of MultiXcan, evaluates the joint association between a gene's predicted expression across multiple tissues and a trait.
Both S-PrediXcan and S-MultiXcan require only GWAS summary statistics, eliminating the need for individual-level genotype and phenotype data.

Here, we provide a concise overview of the TWAS methods pertinent to our regression framework (for more detailed information, refer to the cited articles).
We define $\mathbf{y}$ as a vector of traits for $n$ individuals, which is centered to eliminate the need for an intercept.
The gene's predicted expression for all individuals in tissue $l$ is given by:

\[
\tilde{\mathbf{t}}_l = \sum_{a \in \mathrm{model}_l} w_{a}^{l} X_{a}
\]

where $X_a$ represents the genotype of SNP $a$, and $w_{a}^{l}$ is its corresponding weight in the tissue prediction model $l$.
The standardized version of $\tilde{\mathbf{t}}_l$, denoted as $\mathbf{t}_l$, has a mean of zero and a standard deviation of one.

### Methods

S-PrediXcan [@doi:10.1038/s41467-018-03621-1] is a summary-based version of PrediXcan [@doi:10.1038/ng.3367].
PrediXcan models a trait as a linear function of a gene's expression in a single tissue using the univariate model:

$$
\mathbf{y} = \mathbf{t}_l \gamma_l + \bm{\epsilon}_l,
$$ {#eq:predixcan}

where $\hat{\gamma}_l$ represents the estimated effect size or regression coefficient, and $\bm{\epsilon}_l$ denotes the error terms with variance $\sigma_{\epsilon}^{2}$.
The association's significance is assessed by computing the $z$-score $\hat{z}_{l} = \hat{\gamma}_l / \mathrm{se}(\hat{\gamma}_l)$ for a gene's tissue model $l$.
PrediXcan requires individual-level data to fit this model, whereas S-PrediXcan approximates PrediXcan $z$-scores using only GWAS summary statistics with the expression:

$$
\hat{z}_{l} \approx \sum_{a \in model_{l}} w_a^l \frac{\hat{\sigma}_a}{\hat{\sigma}_l} \frac{\hat{\beta}_a}{\mathrm{se}(\hat{\beta}_a)},
$$ {#eq:spredixcan}

where $\hat{\sigma}_a$ is the variance of SNP $a$, $\hat{\sigma}_l$ is the variance of the predicted expression of a gene in tissue $l$, and $\hat{\beta}_a$ is the estimated effect size of SNP $a$ from the GWAS.
In these TWAS methods, genotype variances and covariances are estimated using the Genotype-Tissue Expression project (GTEx v8) [@doi:10.1126/science.aaz1776] as the reference panel.
Since S-PrediXcan provides tissue-specific directions of effects (e.g., whether higher or lower predicted expression of a gene confers more or less disease risk), we used the $z$-scores in our drug repurposing approach, as described below.

### Revised Paragraph

S-MultiXcan [@doi:10.1371/journal.pgen.1007889] is a summary version of MultiXcan, which is more powerful than PrediXcan for detecting gene-trait associations, although it does not provide the direction of effects.
The main output of MultiXcan is the $p$-value, obtained via an F-test, from the multiple tissue model:

$$
\begin{split}
\mathbf{y} & = \sum_{l=1}^{p} \mathbf{t}_l g_l + \mathbf{e} \\
& = \mathbf{T} \mathbf{g} + \mathbf{e},
\end{split}
$$
{#eq:multixcan}

Here, $\mathbf{T}$ is a matrix with $p$ columns $\mathbf{t}_l$, $\hat{g}_l$ is the estimated effect size for the predicted gene expression in tissue $l$, and $\hat{\mathbf{g}}$ is a vector containing $p$ estimated effect sizes $\hat{g}_l$.
The error terms are denoted by $\mathbf{e}$, with variance $\sigma_{e}^{2}$.
Due to the high correlation between predicted expression values for a gene across different tissues, MultiXcan employs the principal components (PCs) of $\mathbf{T}$ to mitigate collinearity issues.

S-MultiXcan derives the joint regression estimates (effect sizes and their variances) in Equation \ref{eq:multixcan} using the marginal estimates from S-PrediXcan, as defined in Equation \ref{eq:spredixcan}.
Under the null hypothesis of no association, $\hat{\mathbf{g}}^{\top} \frac{\mathbf{T}^{\top}\mathbf{T}}{\sigma_{e}^{2}} \hat{\mathbf{g}} \sim \chi_{p}^{2}$.
The significance of the association in S-MultiXcan is then estimated with:

$$
\begin{split}
\frac{\hat{\mathbf{g}}^{\top} (\mathbf{T}^{\top}\mathbf{T}) \hat{\mathbf{g}}}{\sigma_{e}^{2}} & \approx \bm{\hat{\gamma}}^{\top} \frac{\sqrt{n-1}}{\sigma_{\epsilon}} \left(\frac{\mathbf{T}^{\top} \mathbf{T}}{n-1}\right)^{-1} \frac{\sqrt{n-1}}{\sigma_{\epsilon}} \bm{\hat{\gamma}} \\
& = \hat{\mathbf{z}}^{\top} Cor(\mathbf{T})^{-1} \hat{\mathbf{z}},
\end{split}
$$
{#eq:smultixcan}

In this context, $\hat{\mathbf{z}}$ is a vector containing $p$ $z$-scores (from Equation \ref{eq:spredixcan}) for each tissue available for the gene, and $Cor(\mathbf{T})$ is the autocorrelation matrix of $\mathbf{T}$.
Given that $\mathbf{T}^{\top}\mathbf{T}$ is often singular for many genes, S-MultiXcan calculates the pseudo-inverse $Cor(\mathbf{T})^{+}$ using the top $k$ PCs, leading to $\hat{\mathbf{z}}^{\top} Cor(\mathbf{T})^{+} \hat{\mathbf{z}} \sim \chi_k^2$.
This expression is derived using the conservative approximation $\sigma_{e}^{2} \approx \sigma_{\epsilon}^{2}$, implying that the variance of the error terms in the joint regression approximately equals the residual variance of the marginal regressions.

Additionally, $Cor(\mathbf{T})$ is estimated using a global genotype covariance matrix, whereas marginal $\hat{z}_l$ in Equation \ref{eq:spredixcan} are approximated using tissue-specific genotype covariances.
Although S-MultiXcan provides highly concordant estimates compared to MultiXcan, the results are not perfectly correlated across genes [@doi:10.1371/journal.pgen.1007889].
These differences are significant for our LV-based regression model when computing the gene-gene correlation matrix.
We utilized S-MultiXcan results for our LV-based regression model and our cluster analyses of traits.


### TWAS resources {#sec:methods:twas}

We used two large TWAS resources from different cohorts for discovery and replication, all obtained from European ancestries.
PhenomeXcan [@doi:10.1126/sciadv.aba2083], our discovery cohort, provides results on 4,091 traits across different categories.
Supplementary Data 1 has all the details about the included GWAS, sample size and disease/trait categories.
In PhenomeXcan, these publicly available GWAS summary statistics were used to compute
1) gene-based associations with the PrediXcan family of methods (described before), and
2) a posterior probability of colocalization between GWAS loci and *cis*-eQTL with fastENLOC [@doi:10.1126/sciadv.aba2083; @doi:10.1016/j.ajhg.2020.11.012].
We refer to the matrix of $z$-scores from S-PrediXcan (Equation (@eq:spredixcan)) across $q$ traits and $m$ genes in tissue $t$ as $\mathbf{M}^{t} \in \mathbb{R}^{q \times m}$.
As explained later, matrices $\mathbf{M}^{t}$ were used in our LV-based drug repurposing framework since they provide direction of effects.
The S-MultiXcan results (22,515 gene associations across 4,091 traits) were used in our LV-based regression framework and our cluster analyses of traits.
For the cluster analyses, we used the $p$-values converted to $z$-scores: $\mathbf{M}=\Phi^{-1}(1 - p/2)$, where $\Phi^{-1}$ is the probit function.
Higher $z$-scores correspond to stronger associations.

Our discovery cohort was eMERGE [@doi:10.1038/gim.2013.72], where the same TWAS methods were run on 309 phecodes [@doi:10.1101/2021.10.21.21265225] across different categories (more information about traits are available in [@doi:10.1101/2021.10.21.21265225]).
We used these results to replicate the associations found with our LV-based regression framework in PhenomeXcan.


### MultiPLIER and Pathway-level information extractor (PLIER) {#sec:methods:multiplier}

### Revised Paragraph

MultiPLIER [@doi:10.1016/j.cels.2019.04.003] extracts patterns of co-expressed genes from the recount2 dataset [@doi:10.1038/nbt.3838], excluding GTEx samples.
This approach employs the Pathway-Level Information Extractor (PLIER) method [@doi:10.1038/s41592-019-0456-1], which uses unsupervised learning and canonical pathway knowledge to mitigate technical noise.
PLIER utilizes matrix factorization to decompose gene expression data into latent variables (LVs), where each LV represents a gene module.
The MultiPLIER models reduced the dimensionality of recount2 to 987 LVs.

### Methods

Given a gene expression dataset $\mathbf{Y}^{m \times c}$ with $m$ genes and $c$ experimental conditions, and a prior knowledge matrix $\mathbf{C} \in \{0,1\}^{m \times p}$ for $p$ MSigDB pathways [@doi:10.1016/j.cels.2015.12.004] (where $\mathbf{C}_{ij} = 1$ if gene $i$ belongs to pathway $j$), PLIER identifies matrices $\mathbf{U}$, $\mathbf{Z}$, and $\mathbf{B}$ by minimizing the following objective function:

$$
||\mathbf{Y} - \mathbf{Z}\mathbf{B}||^{2}_{F} + \lambda_1 ||\mathbf{Z} - \mathbf{C}\mathbf{U}||^{2}_{F} + \lambda_2 ||\mathbf{B}||^{2}_{F} + \lambda_3 ||\mathbf{U}||_{L^1}
$$ 
{#eq:met:plier_func}

subject to the constraints $\mathbf{U} > 0$ and $\mathbf{Z} > 0$.
Here, $\mathbf{Z}^{m \times l}$ represents the gene loadings with $l$ latent variables, $\mathbf{B}^{l \times c}$ denotes the latent space for $c$ conditions, and $\mathbf{U}^{p \times l}$ indicates which of the $p$ prior-information pathways in $\mathbf{C}$ are represented for each latent variable.
The parameters $\lambda_i$ are regularization terms used during the training process.

The matrix $\mathbf{Z}$ provides a low-dimensional representation of the gene space, where each latent variable aligns as closely as possible with prior knowledge.
This representation can correspond to either a known or novel gene module (i.e., a meaningful biological pattern) or noise.

For our drug repurposing and cluster analyses, we utilized a model to map gene-trait associations (from TWAS) and gene-drug associations (from LINCS L1000) into a low-dimensional gene module space.
Specifically, TWAS associations $\mathbf{M}$ (derived from either S-PrediXcan or S-MultiXcan) were projected using the following equation:

$$
\hat{\mathbf{M}} = (\mathbf{Z}^{\top} \mathbf{Z} + \lambda_{2} \mathbf{I})^{-1} \mathbf{Z}^{\top} \mathbf{M},
$$
{#eq:proj}

where $\hat{\mathbf{M}}^{l \times q}$ is a matrix representing traits by gene modules instead of individual genes.
As detailed later, we applied the same method to project drug-induced transcriptional profiles from LINCS L1000, thereby obtaining a representation of drugs using gene modules. 

References: [@pmid:33931583; @doi:10.1101/2021.10.21.21265225; @pmid:31036433].


### Regression model for LV-trait associations {#sec:methods:reg}

We adapted the gene-set analysis framework from MAGMA [@doi:10.1371/journal.pcbi.1004219] to TWAS.
We used a competitive test to predict gene-trait associations from TWAS using gene weights from an LV, testing whether top-weighted genes for an LV are more strongly associated with the phenotype than other genes with relatively small or zero weights.
Thus, we fit the model

$$
\mathbf{m}=\beta_{0} + \mathbf{s} \beta_{s} + \sum_{i} \mathbf{x}_{i} \beta_{i} + \bm{\epsilon},
$$ {#eq:reg:model}

### Methods

where $\mathbf{m}$ is a vector of S-MultiXcan gene $p$-values for a trait, transformed by $-log_{10}$; $\mathbf{s}$ is a binary indicator vector with $s_{\ell}=1$ for the top 1% of genes with the highest loadings for latent variable (LV) $\ell$ (from $\mathbf{Z}_{\ell}$) and zero otherwise; $\mathbf{x}_{i}$ represents a gene property used as a covariate; $\beta$ denotes effect sizes, with $\beta_{0}$ as the intercept; and $\bm{\epsilon} \sim \mathrm{MVN}(0, \sigma^{2} \mathbf{R})$ is a vector of error terms following a multivariate normal distribution (MVN), where $\mathbf{R}$ is the matrix of gene correlations.

The model tests the null hypothesis $\beta_{s} = 0$ against the one-sided hypothesis $\beta_{s} > 0$.
In this context, $\beta_{s}$ represents the difference in trait associations between genes included in latent variable (LV) $\ell$ and those not included.
Following the MAGMA framework, we incorporated two gene properties as covariates: 1) *gene size*, defined as the number of principal components (PCs) retained in S-MultiXcan, and 2) *gene density*, defined as the ratio of the number of PCs to the number of tissues available.

References: [@pmid:33931583; @doi:10.1101/2021.10.21.21265225; @pmid:31036433]

### Methods

Since the error terms $\bm{\epsilon}$ could be correlated, we cannot assume they follow independent normal distributions as in a standard linear regression model.
In the PrediXcan family of methods, the predicted expression of a pair of genes could be correlated if they share eQTLs or if these are in linkage disequilibrium (LD) [@doi:10.1038/s41588-019-0385-z].
Therefore, we used a generalized least squares approach to account for these correlations. 

The gene-gene correlation matrix $\mathbf{R}$ was approximated by computing the correlations between the model sum of squares (SSM) for each pair of genes under the null hypothesis of no association.
These correlations are derived from the individual-level MultiXcan model (Equation \ref{eq:multixcan}), where the predicted expression matrix $\mathbf{T}_{i} \in \mathbb{R}^{n \times p_i}$ of a gene $i$ across $p_i$ tissues is projected into its top $k_i$ principal components (PCs), resulting in matrix $\mathbf{P}_{i} \in \mathbb{R}^{n \times k_i}$.
From the MAGMA framework, we know that the SSM for each gene is proportional to $\mathbf{y}^{\top} \mathbf{P}_{i} \mathbf{P}_{i}^{\top} \mathbf{y}$.
Under the null hypothesis of no association, the covariance between the SSM of genes $i$ and $j$ is given by:

$$
\mathrm{Cov}_{\text{SSM}}(i, j) = 2 \times \mathrm{Trace}(\mathbf{P}_{i}^{\top} \mathbf{P}_{j} \mathbf{P}_{j}^{\top} \mathbf{P}_{i}).
$$

The standard deviation of each SSM is given by $\sqrt{2 \times k_{i}} \times (n - 1)$.
Therefore, the correlation between the SSMs for genes $i$ and $j$ can be written as:

$$
\mathbf{R}_{ij} = \frac{2 \times \mathrm{Trace}(\mathbf{P}_{i}^{\top} \mathbf{P}_{j} \mathbf{P}_{j}^{\top} \mathbf{P}_{i})}{\sqrt{2 \times k_{i}} \times \sqrt{2 \times k_{j}} \times (n - 1)^2} = \frac{2 \times \mathrm{Trace}(\mathrm{Cor}(\mathbf{P}_{i}, \mathbf{P}_{j}) \times \mathrm{Cor}(\mathbf{P}_{j}, \mathbf{P}_{i}))}{\sqrt{2 \times k_{i}} \times \sqrt{2 \times k_{j}}}, \label{eq:reg:r}
$$

where the columns of $\mathbf{P}$ are standardized, $\mathrm{Trace}$ denotes the trace of a matrix, and the cross-correlation matrix between PCs $\mathrm{Cor}(\mathbf{P}_{i}, \mathbf{P}_{j}) \in \mathbb{R}^{k_i \times k_j}$ is given by:

$$
\mathrm{Cor}(\mathbf{P}_{i}, \mathbf{P}_{j}) = \mathrm{diag}(\lambda_i)^{-1/2} \mathbf{V}_{i} \left(\frac{\mathbf{T}_{i}^{\top} \mathbf{T}_{j}}{n-1}\right) \mathbf{V}_{j}^{\top} \mathrm{diag}(\lambda_j)^{-1/2}, \label{eq:reg:cor_pp}
$$

where $\frac{\mathbf{T}_{i}^{\top} \mathbf{T}_{j}}{n-1} \in \mathbb{R}^{p_i \times p_j}$ is the cross-correlation matrix between the predicted expression levels of genes $i$ and $j$, and the columns of $\mathbf{V}_{i}$ and scalars $\lambda_i$ are the eigenvectors and eigenvalues of $\mathbf{T}_{i}$, respectively.
S-MultiXcan retains only the top eigenvectors using a condition number threshold of $\frac{\max(\lambda_i)}{\lambda_i} < 30$. 

To estimate the correlation of predicted expression levels for genes $i$ in tissue $k$ and gene $j$ in tissue $l$, $(\mathbf{t}_k^i, \mathbf{t}_l^j)$, we used [@doi:10.1371/journal.pgen.1007889]:

$$
\mathrm{Cor}(\mathbf{t}_k^i, \mathbf{t}_l^j) = \frac{(\mathbf{T}_{i}^{\top} \mathbf{T}_{j})_{kl}}{n-1} = \frac{\mathrm{Cov}(\mathbf{t}_k, \mathbf{t}_l)}{\sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)}} = \frac{\sum_{\substack{a \in \mathrm{model}_k \\ b \in \mathrm{model}_l}} w_a^k w_b^l \Gamma_{ab}}{\sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)}}, \label{eq:reg:corr_genes}
$$

where $X_a$ is the genotype of SNP $a$, $w_a^k$ is the weight of SNP $a$ for gene expression prediction in the tissue model $k$, and $\Gamma = \widehat{\mathrm{var}}(\mathbf{X}) = (\mathbf{X} - \bar{\mathbf{X}})^{\top} (\mathbf{X} - \bar{\mathbf{X}}) / (n-1)$ is the genotype covariance matrix using GTEx v8 as the reference panel, which is the same used in all TWAS methods described here.
The variance of the predicted expression values of gene $i$ in tissue $k$ is estimated as [@doi:10.1038/s41467-018-03621-1]:

$$
\widehat{\mathrm{var}}(\mathbf{t}_k^i) = (\mathbf{W}^k)^\top \Gamma^k \mathbf{W}^k = \sum_{\substack{a \in \mathrm{model}_k \\ b \in \mathrm{model}_k}} w_a^k w_b^k \Gamma_{ab}^k.
\label{eq:reg:var_gene}
$$

### Revised Paragraph

Given that we employed the MultiXcan regression model (Equation \(@eq:multixcan\)), the matrix \(\mathbf{R}\) serves as an approximation of gene correlations within S-MultiXcan.
As previously discussed, S-MultiXcan estimates the joint regression parameters in MultiXcan by using the marginal regression estimates from S-PrediXcan (Equation \(@eq:spredixcan\)) with certain simplifying assumptions and different genotype covariance matrices.
This introduces complexity in deriving an S-MultiXcan-specific solution for computing \(\mathbf{R}\).

To address this, we utilized a submatrix \(\mathbf{R}_{\ell}\), which corresponds only to the genes that are part of latent variable (LV) \(\ell\) (specifically, the top 1% of genes), rather than the entire matrix \(\mathbf{R}\).
This conservative approach ensures that correlations are considered for the top genes only.
Our simulations ([Supplementary Note 1](#sm:reg:null_sim)) demonstrate that the model is approximately well-calibrated and capable of correcting for LVs with adjacent and highly correlated top genes (e.g., Figure \(@fig:reg:nulls:qqplot:lv234\)). 

However, the simulation also identified 127 LVs where the model was not well-calibrated (e.g., Figure \(@fig:reg:nulls:qqplot:lv914\)).
This issue is likely due to challenges in accurately computing the gene correlation matrix.
Consequently, we excluded these LVs from our main analyses.

In Equation (1), for each gene, we only considered tissue models present in S-PrediXcan results and SNPs present in the GWAS used as input for the TWAS approaches.
This was necessary to obtain more accurate correlation estimates [@doi:10.1371/journal.pgen.1007889].
Consequently, we computed different correlation matrices for PhenomeXcan and eMERGE.
In PhenomeXcan, most of the GWAS (4,049) were obtained from the UK Biobank using the same pipeline and included the same set of SNPs, so a single correlation matrix was used for this set.
For the remaining GWAS, we utilized a single correlation matrix for each group of traits that shared the same or most of the SNPs.

We ran our regression model for all 987 LVs across the 4,091 traits in PhenomeXcan.
For replication, we ran the model in the 309 phecodes in eMERGE.
We adjusted the $p$-values using the Benjamini-Hochberg procedure.


### LV-based drug repurposing approach {#sec:methods:drug}

For the drug-disease prediction, we developed a latent variable (LV)-based method within a drug repositioning framework previously used for psychiatric traits [@doi:10.1038/nn.4618].
In this framework, individual genes associated with a trait are anticorrelated with drug expression profiles.
We compared our LV-based method with this previously published single-gene approach.

In the single-gene method, we calculated a drug-disease score by multiplying each set of signed $z$-scores from S-PrediXcan in tissue $t$, denoted as $\mathbf{M}^t$, with another set of signed $z$-scores from transcriptional responses profiled in LINCS L1000 [@doi:10.1016/j.cell.2017.10.049], represented as $\mathbf{L}^{c \times m}$ (for $c$ compounds).
The matrix $\mathbf{M}^t$ contains information on whether higher or lower predicted gene expression is associated with disease risk, while $\mathbf{L}$ indicates whether a drug increases or decreases gene expression.
Thus, these two matrices can be multiplied to compute a drug-disease score.

The resulting product is given by:

$$
\mathbf{D}^{t,k} = -1 \cdot \mathbf{M}^{t,k} \mathbf{L}^\top
$$

where $k$ refers to the number of most significant gene associations in $\mathbf{M}^t$ for each trait.
Following the suggestion in [@doi:10.1038/nn.4618], $k$ could include all genes or the top 50, 100, 250, and 500 genes.
We then averaged the score ranks across all $k$ to obtain $\mathbf{D}^t$.
Finally, for each drug-disease pair, we selected the maximum prediction score across all tissues:

$$
\mathbf{D}_{ij} = \max \{ \mathbf{D}_{ij}^t \mid \forall t \}
$$


The same procedure was applied to the latent variable (LV)-based approach.
We projected $\mathbf{M}^{t}$ and $\mathbf{L}$ into the gene module latent space using Equation (@eq:proj), resulting in $\hat{\mathbf{M}}^t$ and $\hat{\mathbf{L}}^{l \times c}$, respectively.
Finally, $\mathbf{D}^{t,k} = -1 \cdot \hat{\mathbf{L}}^{\top} \hat{\mathbf{M}}^{t,k}$, where $k$ could represent either all LVs or the top 5, 10, 25, and 50 LVs (since the number of LVs is an order of magnitude less than the number of genes).


Since the gold standard of drug-disease medical indications is described with Disease Ontology IDs (DOID) [@doi:10.1093/nar/gky1032], we mapped PhenomeXcan traits to the Experimental Factor Ontology [@doi:10.1093/bioinformatics/btq099] using [@url:https://github.com/EBISPOT/EFO-UKB-mappings], and then to DOID.


### Consensus clustering of traits {#sec:methods:clustering}

### Revised Paragraph

We conducted two preprocessing steps on the S-MultiXcan results prior to the cluster analysis.
First, we combined the results in $\mathbf{M}$ (with $p$-values converted to $z$-scores, as previously described) for traits that mapped to the same Experimental Factor Ontology (EFO) term [@doi:10.1093/bioinformatics/btq099] using Stouffer's method:

$$
\sum w_i M_{ij} / \sqrt{\sum w_i^2}
$$

where $w_i$ is a weight based on the GWAS sample size for trait $i$, and $M_{ij}$ is the $z$-score for gene $j$.
Second, to mitigate the influence of highly polygenic traits, we normalized the $z$-scores for each trait $i$ by dividing them by their sum:

$$
M_{ij} / \sum M_{ij}
$$

Finally, we projected this data matrix using Equation (@eq:proj), resulting in $\hat{\mathbf{M}}$ with $n = 3,752$ traits and $l = 987$ latent variables (LVs) as the input for our clustering pipeline.


### Revised Paragraph:

A partitioning of $\hat{\mathbf{M}}$ with $n$ traits into $k$ clusters is represented by a label vector $\pi \in \mathbb{N}^n$.
Consensus clustering involves two main steps: 

1.
**Generation of an Ensemble**: Create an ensemble $\Pi$ consisting of $r$ partitions of the dataset: $\Pi = \{\pi_1, \pi_2, \ldots, \pi_r\}$.
2.
**Combination of the Ensemble**: Combine the ensemble into a consolidated solution defined as:

    $$
    \pi^* = \mathrm{arg}\,\underset{\hat{\pi}}{\max} Q(\{ \lvert \mathcal{L}^i \lvert \phi(\hat{\pi}_{\mathcal{L}^i}, \pi_{i \mathcal{L}^i}) \mid i \in \{1,\ldots,r\} \}),
    $$ {#eq:consensus:obj_func}

    where $\mathcal{L}^i$ is a set of data indices with known cluster labels for partition $i$, $\phi\colon \mathbb{N}^n \times \mathbb{N}^n \to \mathbb{R}$ is a function that measures the similarity between two partitions, and $Q$ is a measure of central tendency, such as the mean or median.
We used the adjusted Rand index (ARI) [@doi:10.1007/BF01908075] for $\phi$ and the median for $Q$.

To obtain $\pi^*$, we define a consensus function $\Gamma\colon \mathbb{N}^{n \times r} \to \mathbb{N}^n$ with $\Pi$ as the input.
We utilized consensus functions based on the evidence accumulation clustering (EAC) paradigm [@doi:10.1109/TPAMI.2005.113].
In this approach, $\Pi$ is first transformed into a distance matrix $\mathbf{D}_{ij} = d_{ij} / r$, where $d_{ij}$ is the number of times traits $i$ and $j$ were grouped in different clusters across all $r$ partitions in $\Pi$.
Subsequently, $\Gamma$ can be any similarity-based clustering algorithm, which is applied to $\mathbf{D}$ to derive the final partition $\pi^*$.


### Methods

For the ensemble generation step, we employed various algorithms to create a highly diverse set of partitions (see Figure @fig:clustering:design), as diversity is crucial for ensembles [@doi:10.1016/j.ins.2016.04.027; @doi:10.1109/TPAMI.2011.84; @doi:10.1016/j.patcog.2014.04.005].
We utilized three data representations: the raw dataset, its projection into the top 50 principal components, and the embedding learned by UMAP [@arxiv:1802.03426] using 50 components.

For each data representation, we applied five clustering algorithms, each with different assumptions about the data structure: $k$-means [@Arthur2007], spectral clustering [@Ng2001], a Gaussian mixture model (GMM), hierarchical clustering, and DBSCAN [@Ester1996].
For $k$-means, spectral clustering, and GMM, we specified a range of $k$ values from 2 to $\sqrt{n} \approx 60$, generating five partitions for each $k$ using random seeds.
For hierarchical clustering, we produced four partitions per $k$ using common linkage criteria: ward, complete, average, and single.

For DBSCAN, we combined different ranges for the parameters $\epsilon$ (the maximum distance between two data points to be considered part of the same neighborhood) and *minPts* (the minimum number of data points in a neighborhood for a data point to be considered a core point), following the procedure in [@doi:10.1088/1755-1315/31/1/012012].
Specifically, we used *minPts* values from 2 to 125.
For each data representation (raw, PCA, and UMAP), we determined a plausible range of $\epsilon$ values by observing the distribution of the mean distance of the *minPts*-nearest neighbors across all data points. 

Since some combinations of *minPts* and $\epsilon$ might not produce meaningful partitions (e.g., when all points are detected as noisy or only one cluster is found), we resampled partitions generated by DBSCAN to ensure equal representation of this algorithm in the ensemble.
This procedure resulted in a final ensemble of 4,428 partitions of 3,752 traits.


Finally, we applied spectral clustering on $\mathbf{D}$ to obtain the final consensus partitions.
We first transformed $\mathbf{D}$ into a similarity matrix using the Radial Basis Function (RBF) kernel, $\mathrm{exp}(-\gamma \mathbf{D}^2)$, with four empirically determined values for $\gamma$.
For each $k$ ranging from 2 to 60, we generated four consensus partitions and selected the one that maximized Equation (@eq:consensus:obj_func).
We then filtered these 59 solutions to retain only those with ensemble agreement above the 75th percentile (Figure @fig:sup:clustering:agreement), resulting in 15 final consensus partitions, as illustrated in Figure @fig:clustering:tree.

### Revised Paragraph:

The input data in our clustering pipeline undergo several linear and nonlinear transformations, including Principal Component Analysis (PCA), Uniform Manifold Approximation and Projection (UMAP), and the ensemble transformation using the Evidence Accumulation Clustering (EAC) paradigm (distance matrix $\mathbf{D}$).
Although consensus clustering has clear advantages for biological data [@pmid:27303057], this series of data transformations complicates the interpretation of results.
To address this issue, we employed a supervised learning approach to identify which gene modules/latent variables (LVs) are most important for each cluster of traits (Figure {@fig:clustering:design}b).
It is important to note that this supervised model was not used for prediction but solely to determine which features (LVs) were most discriminative for each cluster.

For this purpose, we used the highest resolution partition ($k$ = 29, although any partition could be used) to train a decision tree model.
Each cluster served as a label, and the projected data $\hat{\mathbf{M}}$ were used as the training samples.
For each $k$, we created a set of binary labels with the current cluster's traits as the positive class and the rest of the traits as the negative class.
We then selected the LV at the root node of the trained model only if its threshold was positive and exceeded one standard deviation.
Subsequently, we removed this LV from $\hat{\mathbf{M}}$ (regardless of whether it had been previously selected) and retrained the model.
This procedure was repeated 20 times to extract the top 20 LVs that best discriminate traits in a cluster from the rest.

In [Supplementary Note 2](#sm:clustering:null_sim), we performed several analyses under a null hypothesis of no structure in the data to verify that the clustering results detected by this pipeline were real.


### CRISPR-Cas9 screening {#sec:methods:crispr}

### Cell Culture

HepG2 cells were obtained from ATCC (ATCC® HB-8065™) and cultured in Eagle's Minimum Essential Medium with L-Glutamine (EMEM, Cat.
112-018-101, Quality Biology).
The medium was supplemented with 10% Fetal Bovine Serum (FBS, Gibco, Cat.
16000-044) and 1% Penicillin-Streptomycin (Pen/Strep, Gibco, Cat.
15140-122).
Cells were incubated at 37°C in a humidity-controlled environment with 5% CO2.
They were maintained in Collagen-I coated flasks at a density not exceeding 80% confluency.

### Genome-wide Lentiviral Pooled CRISPR-Cas9 Library

The third-generation lentiviral, Broad GPP genome-wide Human Brunello CRISPR knockout pooled library, provided by David Root and John Doench from Addgene (Cat.
73179-LV), was utilized for HepG2 cell transduction.
This library comprises 76,441 single guide RNAs (sgRNAs) targeting 19,114 human genes, with an average of four sgRNAs per gene.
Each 20-nucleotide (nt) sgRNA cassette was inserted into the lentiCRISPRv2 backbone between the U6 promoter and the gRNA scaffold.

During cell transduction, lentiviral vectors encoding Cas9 were used to deliver the sgRNA cassette-containing plasmids into cells.
Cells that were not successfully transduced were excluded through puromycin selection.

**Lentiviral Titer Determination**

No-spin lentiviral transduction was employed for the screen.
Approximately 2.5 million cells were seeded per well in a Collagen-I coated 6-well plate, in the presence of 8 µg/ml polybrene (Millipore Sigma, Cat.
TR-1003 G).
Various volumes of titrated virus (e.g., 0, 50, 100, 200, 250, and 400 µl) were added to each well.
EMEM complete media was added to achieve a final volume of 1.24 ml.
Sixteen to eighteen hours post-transduction, the virus/polybrene-containing media was removed.
Cells were washed twice with 1x DPBS and fresh EMEM was added.
At 24 hours post-transduction, cells in each well were trypsinized, diluted (e.g., 1:10), and seeded into pairs of wells in new 6-well plates. 

At 60 hours post-transduction, the media in each well was replaced with fresh EMEM.
Puromycin (2 µg/ml, Gibco, Cat.
A1113803) was added to one well of each pair.
Two to five days after puromycin selection, or when no cells survived in the 0 virus well treated with puromycin, cells in both wells (with and without puromycin) were collected and counted for viability.
The percentage of infection (PI%) was calculated by comparing the cell numbers with and without puromycin selection within each pair.

Using Poisson's distribution theory, when the transduction efficiency (PI%) is between 30-50%, it corresponds to a multiplicity of infection (MOI) of approximately 0.35-0.70.
At an MOI close to 0.3, around 25% of cells are infected, with most infected cells predicted to have only one copy of the virus.
Therefore, a virus volume of 120 µl, yielding a transduction efficiency of 30-40%, was selected for further large-scale viral transduction.

References: [@pmid:33931583; @doi:10.1101/2021.10.21.21265225; @pmid:31036433]

**Lentiviral Transduction in HepG2 Using Brunello CRISPR Knockout Pooled Library**

To ensure a coverage of at least 500 cells per single-guide RNA (sgRNA) and a multiplicity of infection (MOI) between 0.3 and 0.4—ensuring that 95% of infected cells receive only one viral particle—approximately 200 million cells were initiated for the screen.
The transduction process followed previously described methods.
Briefly, 2.5 million cells were seeded in each well of fourteen 6-well plates, with 8 µg/mL of polybrene.
A volume of 120 µL of the virus was added to each experimental well. 

Eighteen hours post-transduction, the virus/polybrene mixture was removed, and cells from each well were collected, counted, and pooled into T175 flasks.
Sixty hours post-transduction, 2 µg/mL of puromycin was added to each flask.
The medium was changed every two days with fresh EMEM supplemented with 2 µg/mL puromycin.
Seven days after puromycin selection, cells were collected, pooled, counted, and replated.

**Fluorescent Dye Staining**

Nine days post-puromycin selection, cells were divided into two groups.
A sample of 20-30 million cells was collected as the Unsorted Control.
The cell pellet was centrifuged at 500 x g for 5 minutes at 4°C.
The dry pellet was stored at -80°C for subsequent genomic DNA isolation.
The remaining cells (approximately 200 million) were maintained in 100 mm dishes and stained with a fluorescent dye (LipidSpot™ 488, Biotium, Cat.
70065-T).
Briefly, LipidSpot 488 was diluted 1:100 with DPBS.
Four milliliters of the staining solution were added to each dish and incubated at 37°C for 30 minutes.
Cell images were captured using an EVOS fluorescent microscope to detect the GFP signal (Figure @fig:sup:crispr:fig1).

**Fluorescence-Activated Cell Sorting (FACS)**

Cells were immediately collected into 50 mL tubes and kept cold throughout the process.
The cells were centrifuged at 500 × g for 5 minutes at 4°C.
After washing with DPBS, the cell pellets were resuspended in FACS Sorting Buffer (1× DPBS without Ca²⁺/Mg²⁺, 2.5 mM EDTA, 25 mM HEPES, 1% BSA).
This solution was filter-sterilized and maintained at 4°C.
Gentle pipetting was used to ensure a single-cell suspension.
The cell solution was then filtered through a cell strainer (Falcon, Cat.
352235) and kept on ice, protected from light.

Cells were sorted using a FACSJazz instrument with a 100 μm nozzle.
Approximately 20% of both GFP-High and GFP-Low populations (Figure @fig:sup:crispr:fig2) were collected into 15 mL tubes.
Post-sorting, the cells were immediately centrifuged, and the resulting pellets were stored at -80°C for subsequent genomic DNA isolation.

**Genomic DNA Isolation and Verification**

Genomic DNA was extracted from three conditions: Un-Sorted Control, lentiV2 GFP-High, and lentiV2 GFP-Low, using the QIAamp DNA Blood Mini Kit (Qiagen, Cat.
51104).
The quality and quantity of the genomic DNA (gDNA) were assessed via UV spectroscopy using a Nanodrop spectrophotometer.
For each condition, 80-160 µg of gDNA was isolated.
Verification of the sgRNA cassette and lentiviral-specific transgene in the isolated gDNA was performed using PCR (Figure @fig:sup:crispr:fig3).

References:
[@pmid:33931583; @doi:10.1101/2021.10.21.21265225; @pmid:31036433]

### Illumina Libraries Generation and Sequencing

The fragment containing the sgRNA cassette was amplified using P5/P7 primers, as described in [@pmid:26780180].
Primer sequences were adapted from the Broad Institute protocol (Figure @fig:sup:crispr:table1).
A stagger sequence (0-8 nt) was included in P5, and an 8 bp unique barcode sequence was included in P7.
Primers were synthesized by Integrated DNA Technologies (IDT) and each primer was PAGE purified.

For each condition, 32 PCR reactions were set up.
Each 100 µL PCR reaction contained approximately 5 µg of gDNA, 5 µL of each 10 µM P5 and P7 primers, and ExTaq DNA Polymerase (TaKaRa, Cat.
RR001A).
The thermal cycler parameters were as follows: initial denaturation at 95°C for 1 min; 24 cycles of denaturation at 94°C for 30 s, annealing at 52.5°C for 30 s, and extension at 72°C for 30 s; followed by a final elongation at 72°C for 10 min.
The expected PCR products were 285-293 bp in length (Figure @fig:sup:crispr:fig4 A).

PCR products from the same condition were pooled and purified using SPRIselect beads (Beckman Coulter, Cat.
B23318).
The purified Illumina libraries were quantified using Qubit, and the quality was assessed with a Bioanalyzer using a High Sensitivity DNA Chip.
A single peak at approximately 285 bp was expected (Figure @fig:sup:crispr:fig4 B).

The final Illumina library samples were sequenced on a NovaSeq 6000.
Samples were pooled and loaded onto an SP flow cell, with a 20% PhiX control v3 library spike-in.


## Data availability

All the main datasets generated in this study are available at [https://doi.org/10.5281/zenodo.8071382](https://doi.org/10.5281/zenodo.8071382) [@doi:10.5281/zenodo.8071382] and the GitHub repository [https://github.com/greenelab/phenoplier](https://github.com/greenelab/phenoplier).

The main input datasets used are TWAS from PhenomeXcan [@doi:10.1126/sciadv.aba2083] for 4,091 traits and from the Electronic Medical Records and Genomics (eMERGE) network phase III [@doi:10.1101/2021.10.21.21265225] for 309 traits; transcriptional responses to small molecule perturbations from LINCS L1000 [@doi:10.1016/j.cell.2017.10.049] that were further preprocessed and mapped to DrugBank IDs from [@doi:10.5281/zenodo.47223]; latent space/gene module models from MultiPLIER [@doi:10.1016/j.cels.2019.04.003].

The data utilized from PhenomeXcan, LINCS L1000, and MultiPLIER are publicly accessible.
Significant results for the eMERGE and Penn Medicine BioBank (PMBB) phenome-wide TWAS are detailed in [@doi:10.1101/2021.10.21.21265225].
Due to institutional privacy policies, the individual-level PMBB raw datasets cannot be made publicly available.
For data access requests, please contact the Penn Medicine Biobank at [https://pmbb.med.upenn.edu/pmbb/](https://pmbb.med.upenn.edu/pmbb/).
The eMERGE network phase III data can be accessed via dbGAP (Accession: phs001584.v2.p2).


## Code availability

The code necessary to reproduce all the analyses in this work is available at [https://doi.org/10.5281/zenodo.8071382](https://doi.org/10.5281/zenodo.8071382) [@doi:10.5281/zenodo.8071382] and the GitHub repository [https://github.com/greenelab/phenoplier](https://github.com/greenelab/phenoplier).

### Methods: Data Analysis Tools and Software

For the CRISPR screening, we utilized FlowJo v10.7 and FACS Jazz Software v1.1.
Data analysis was conducted using Python 3.8 and R 3.6, incorporating several computational packages.
The primary Python packages employed were:

- Jupyter Lab (2.2)
- pandas (1.1)
- matplotlib (3.3)
- seaborn (0.11)
- numpy (1.19)
- scipy (1.5)
- scikit-learn (0.23)
- umap-learn (0.4)

The main R packages used included:

- Bioconductor (3.10)
- clusterProfiler (3.14)
- clustree (0.4)
- fgsea (1.17)

We developed several scripts and notebooks, which are available under an open-source license.
All necessary steps for the analyses are thoroughly documented.
Additionally, we provide a Docker image to replicate the runtime environment used in our study, along with a demo to facilitate quick testing of the methods on real data.
